extensions/micrometer/deployment/src/test/java/io/quarkus/micrometer/deployment/binder/UriTagTest.java

```diff
@@ -0,0 +1,92 @@
+        when().get("/ping/one").then().statusCode(200);
+        when().get("/ping/two").then().statusCode(200);
+        when().get("/ping/three").then().statusCode(200);
+        when().get("/vertx/item/123").then().statusCode(200);
+        when().get("/vertx/item/1/123").then().statusCode(200);
+        when().get("/servlet/12345").then().statusCode(200);
+
+        System.out.println("Server paths\n" + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+        System.out.println("Client paths\n" + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+
+        // /one should map to /two, which is ignored. Neither should exist w/ timers
+        Assertions.assertEquals(0, registry.find("http.server.requests").tag("uri", "/one").timers().size(),
+                "/one is mapped to /two, which should be ignored. Found:\n"
+                        + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+        Assertions.assertEquals(0, registry.find("http.server.requests").tag("uri", "/two").timers().size(),
+                "/two should be ignored. Found:\n"
+                        + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+
+        // URIs for server: /ping/{message}, /pong/{message}, /vertx/item/{id}, /vertx/item/{id}/{sub}, /servlet/
+        Assertions.assertEquals(1, registry.find("http.server.requests").tag("uri", "/ping/{message}").timers().size(),
+                "/ping/{message} should be returned by JAX-RS. Found:\n"
+                        + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+        Assertions.assertEquals(1, registry.find("http.server.requests").tag("uri", "/pong/{message}").timers().size(),
+                "/pong/{message} should be returned by JAX-RS. Found:\n"
+                        + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+        Assertions.assertEquals(1, registry.find("http.server.requests").tag("uri", "/vertx/item/{id}").timers().size(),
+                "Vert.x Web template path (/vertx/item/:id) should be detected/translated to /vertx/item/{id}. Found:\n"
+                        + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+        Assertions.assertEquals(1, registry.find("http.server.requests").tag("uri", "/vertx/item/{id}/{sub}").timers().size(),
+                "Vert.x Web template path (/vertx/item/:id/:sub) should be detected/translated to /vertx/item/{id}/{sub}. Found:\n"
+                        + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+
+        Assertions.assertEquals(1, registry.find("http.server.requests").tag("uri", "/servlet").timers().size(),
+                "Servlet path (/servlet) should be used for servlet. Found:\n"
+                        + Util.listMeters(registry.find("http.server.requests").meters(), "uri"));
+
+        // TODO: Fixing in master #15047
```

Comment:
Is this assertion meant to be uncommented as it refers to this PR?

Reply:
I'll change the # in the comment. I don't want to remove it, because it comes next. But in trying to trim down what this was doing, I didn't get to adding the client fixes